java/team25core/OpticalDistanceSensorCriteria.java

```diff
@@ -44,6 +47,9 @@ public void setThreshold(double percent) {
     public boolean satisfied()
     {
         RobotLog.i("251 Light: %f, Threshold: %f", sensor.getRawLightDetected(), threshold);
+        robot.telemetry.addData("Light:", sensor.getRawLightDetected());
+        robot.telemetry.addData("Threshold:", threshold);
+
```

Comment:
This, being a general purpose infrastructure class, I'd like to keep it clean.  Please revert the addition of the Robot parameter to the constructor and remove these two lines.

The robot log really is sufficient here and gives you better data anyway.  You can however remove the 251 tag from the robot log output if you like.